### PR TITLE
BUG: GH11880 where __contains__ fails in unpacked DataFrame with object cols

### LINKED ISSUE #11880: Msgpack - ValueError: buffer source array is read-only
I get the Value error when processing data using pandas. I followed the following steps:
1.  convert to msgpack format with compress flag
2. subsequently read file into a dataframe
3. push to sql table with to_sql

On the third step i get ValueError: buffer source array is read-only.

This problem does not arise if I wrap the read_msgpack call inside a pandas.concat

Example
```python

import pandas as pd
import numpy as np

from sqlalchemy import create_engine

eng = create_engine("sqlite:///:memory:")

df1 = pd.DataFrame({ 'A' : 1.,
                                   'B' : pd.Timestamp('20130102'),
                                   'C' : pd.Series(1,index=list(range(4)),dtype='float32'),
                                   'D' : np.array([3] * 4,dtype='int32'),
                                   'E' : 'foo' })

df1.to_msgpack('test.msgpack', compress='zlib')
df2 = pd.read_msgpack('test.msgpack')

df2.to_sql('test', eng, if_exists='append', chunksize=1000) # throws value error

df2 = pd.cooncat([pd.read_msgpack('test.msgpack')])

df2.to_sql('test', eng, if_exists='append', chunksize=1000) # works
```

This happens with both blosc and zlib compression. While I have found a solution, this behaviour seems very odd and for very large files there is a small performance hit.

edit: @TomAugspurger changed the sql engine to sqlite

### PR DESCRIPTION
**Pull request by kawochen**
closes #11880

### CODE REVIEW

File: pandas/io/tests/test_packers.py [hunk 1]
```diff
@@ -23,7 +24,12 @@
 
 nan = np.nan
 
-from pandas.io.packers import to_msgpack, read_msgpack
+try:
+    import blosc
+except ImportError:
+    _BLOSC_INSTALLED = False
```
Comment (jreback) [inline pandas/io/tests/test_packers.py]: can u do this for ``zlib`` as well

**Comment by jreback** [comment]
can you add a simplified version of the test in #11880 (better than the check on the readability of the internals, but if you can't make a better test, use that)

File: pandas/io/tests/test_packers.py (hunk 1 above)
Comment (kawochen) [inline pandas/io/tests/test_packers.py]: `zlib` is in the standard library

Comment (jreback) [inline pandas/io/tests/test_packers.py]: I know, but it doesn't actually *have* to build with python (e.g if you build it yourself).

Comment (kawochen) [inline pandas/io/tests/test_packers.py]: Oh I see

Comment (jreback) [inline pandas/io/tests/test_packers.py]: not a big deal, really just for consistency (say we add more compressors)

**Comment by kawochen** [comment]
updated

**Comment by jreback** [comment]
thanks!

**Pull request merged**